hearvalidator/validate.py

```diff
@@ -46,6 +43,23 @@ def __init__(self, module_name: str, model_file_path: str, device: str = None):
     def __call__(self):
         self.import_model()
         self.check_load_model()
+
+        # Perform validation. If a tensorflow model was loaded and a specific
+        # device was specified, then reload the model on the correct device.
+        # This is a bit awkward but I wanted to avoid all the tensorflow initialization
+        # stuff before loading a module, which could potentially be a PyTorch module.
+        # So we only get into the tensorflow device stuff if we found a tf module.
```

Comment:
This whole device argument piece makes things a bit awkward when working with both tensorflow and pytorch models. One option would be to add another positional arg for the framework: "torch" or "tf". And then only import the framework that is required. That could simplify things. That being said, this is working fine for now.

Reply:
Where is the current self.model_type thing documented / called by the user?

Reply:
Maybe you can explain to me better about device differences in pytorch and tf

Reply:
The model_type isn't called by the user -- it is inferred based on the type of object returned by load_model. This will probably need to be updated to accept other objects like a kera.Model, etc... Alternative would be to change this to a command line arg and only import the required framework.

Reply:
https://github.com/neuralaudio/hear-validator/issues/4